idseq_pipeline/commands/non_host_alignment_functions.py

```diff
@@ -1125,9 +1108,7 @@ def thread_succeeded(name):
     t_annotation.join()
     assert thread_succeeded("annotation")
 
-    for thread_name, t in uploader_threads.items():
-        t.join()
-        assert thread_succeeded(thread_name), "thread {} failed".format(thread_name)
+    async_handler.wait_on_all()
```

Comment:
You are just being polite here.  I like it.   But if you skipped this, it would happen by default.  The main program would just wait for all threads.